cpp/src/arrow/testing/fixed_width_test_util.h

```diff
@@ -35,24 +29,80 @@ class NestedListGenerator {
   ///
   /// \return `fixed_size_list(fixed_size_list(..., sizes[1]), sizes[0])`
   static std::shared_ptr<DataType> NestedFSLType(
-      const std::shared_ptr<DataType>& inner_type, const std::vector<int>& sizes) {
-    auto type = inner_type;
-    for (auto it = sizes.rbegin(); it != sizes.rend(); it++) {
-      type = fixed_size_list(std::move(type), *it);
-    }
-    return type;
-  }
+      const std::shared_ptr<DataType>& inner_type, const std::vector<int>& sizes);
 
   /// \brief Create a nested FixedListType.
```

Comment:
You should move these template functions to an anonymous namespace in the `.cc` file.